### ISSUE TITLE
OAUTHv2/OpenID Leakage of Authorization Code in URL Query

### ISSUE THREAD

**Issue by halfluke** [akabe1/OAUTHScan#9]
Isn't it normal to have a code in the URL when the Authorization Code flow is used?

Unless I misinterpret the meaning of the finding.

Thank you.

**Comment by akabe1** [comment]
Although this is the behavior expected by the standard, it presents a security flaw and in fact the PKCE  solution was introduced. 
I kindly ask you in the future to open issues for problems on the OAuthScan tool, and not to ask about topics you can find by reading the online RFCs or references.

**Comment by halfluke** [comment]
Well, I thought it might be a false positive but I also specified that I may miss something. With your answer I have now checked how PKCE works and now I really think it should not be reported unless PKCE is also confirmed not being used. PKCE does not prevent a code from being exchanged in the URL query, it just makes stealing the auth code useless.
The issue would still be flagged even when PKCE is used. Perhaps the idea here is to "Use form post response mode instead of redirect for the authorization response"? The extension is useful but seems to flag any theoretical vulnerability without considering specific combinations. Also, closing a question with such a quick and insufficient answer is not nice.
